saq/queue.py

```diff
@@ -370,9 +370,9 @@ async def finish(self, job, status, *, result=None, error=None):
         async with self.redis.pipeline(transaction=True) as pipe:
             pipe = pipe.lrem(self._active, 1, job_id).zrem(self._incomplete, job_id)
 
-            if job.ttl:
+            if job.ttl > 0:
                 pipe = pipe.setex(job_id, job.ttl, self.serialize(job))
-            else:
+            elif job.ttl == 0:
```

Comment:
Should we be deleting the key if job.ttl < 0? 

We store the serialized job when enqueueing it, so doing nothing might actually leave the job stored forever.

Reply:
yea, we need to explicitly delete the job

Reply:
Alright, I fixed it